File: src/main/java/com/github/cao/awa/conium/datapack/inject/item/action/handler/math/typed/IntegerNumberHandler.java
package com.github.cao.awa.conium.datapack.inject.item.action.handler.math.typed;

import com.github.cao.awa.conium.datapack.inject.item.action.ItemPropertyInjectAction;

public class IntegerNumberHandler extends NumberHandler<Integer> {
    @Override
    public Integer doHandle(Integer first, Integer second, ItemPropertyInjectAction action) {
        return switch (action) {
            case ADD -> first + second;
            case MINUS -> first - second;
            case DIVIDE -> first / second;
            case MULTIPLY -> first * second;
            default -> first;
        };
    }
}

File: common/src/main/java/com/github/cao/awa/conium/datapack/inject/item/action/handler/math/typed/IntegerNumberHandler.kt
package com.github.cao.awa.conium.datapack.inject.item.action.handler.math.typed

import com.github.cao.awa.conium.datapack.inject.item.action.ItemPropertyInjectAction

class IntegerNumberHandler : NumberHandler<Int>() {
    override fun doHandle(first: Int, second: Int, action: ItemPropertyInjectAction): Int {
        return when (action) {
            ItemPropertyInjectAction.ADD -> first + second
            ItemPropertyInjectAction.MINUS -> first - second
            ItemPropertyInjectAction.DIVIDE -> first / second
            ItemPropertyInjectAction.MULTIPLY -> first * second
            else -> first
        }
    }
}
